src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java

```diff
@@ -227,20 +227,25 @@ private Runnable lockStudentRepositories(ProgrammingExercise exercise, Predicate
             SecurityUtils.setAuthorizationObject();
             try {
                 List<ProgrammingExerciseStudentParticipation> failedLockOperations = removeWritePermissionsFromAllStudentRepositories(programmingExerciseId, condition);
+                // Stash also the not submitted/committed changes, to ensure that only submitted/commited changes are displayed
+                List<ProgrammingExerciseStudentParticipation> failedStashOperations = stashChangesInAllStudentRepositories(programmingExerciseId, condition);
 
-                // We sent a notification to the instructor about the success of the repository locking operation.
+                // We sent a notification to the instructor about the success of the repository locking and stashing operations.
                 long numberOfFailedLockOperations = failedLockOperations.size();
+                long numberOfFailedStashOperations = failedStashOperations.size();
                 Optional<ProgrammingExercise> programmingExercise = programmingExerciseRepository.findWithTemplateParticipationAndSolutionParticipationById(programmingExerciseId);
                 if (programmingExercise.isEmpty()) {
                     throw new EntityNotFoundException("programming exercise not found with id " + programmingExerciseId);
                 }
-                if (numberOfFailedLockOperations > 0) {
+                if (numberOfFailedLockOperations > 0 || numberOfFailedStashOperations > 0) {
                     groupNotificationService.notifyInstructorGroupAboutExerciseUpdate(programmingExercise.get(),
-                            Constants.PROGRAMMING_EXERCISE_FAILED_LOCK_OPERATIONS_NOTIFICATION + failedLockOperations.size());
+                            Constants.PROGRAMMING_EXERCISE_FAILED_LOCK_OPERATIONS_NOTIFICATION + numberOfFailedLockOperations
+                                    + ". Furthermore, When stashing the not submitted changes, not all stash operations were successfull. Number of failed stash operations: "
+                                    + numberOfFailedStashOperations);
```

Comment:
Reformulate / adjust reporting logic to notify only about operations that failed (not always both).